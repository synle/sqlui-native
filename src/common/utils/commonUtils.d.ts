export declare function getGeneratedRandomId(prefix: string): string;
