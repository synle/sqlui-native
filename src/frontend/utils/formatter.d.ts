export declare const formatSQL: (val: string) => string;
export declare const formatJS: (val: string) => string;
export declare const formatDuration: (durationMs: number) => string;
export declare const escapeSQLValue: (value?: string) => string;
export declare const isValueNumber: (value: any) => boolean;
export declare const isValueBoolean: (value: any) => boolean;
