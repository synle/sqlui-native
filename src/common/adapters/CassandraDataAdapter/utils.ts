import BaseDataAdapter from 'src/common/adapters/BaseDataAdapter/index';

type ClientOptions = {
  host: string;
  port: number;
  contactPoints: string[];
  keyspace?: string;
  authProvider?: {
    username: string;
    password: string;
  };
};

export function getClientOptions(connectionOption: string, database?: string): ClientOptions {
  const connectionParameters = BaseDataAdapter.getConnectionParameters(connectionOption);

  const connectionHosts = connectionParameters?.hosts || [];
  if (connectionHosts.length === 0) {
    // we need a host in the connection string
    throw 'Invalid connection. Host and Port not found';
  }

  const host = connectionHosts[0].host;
  const port = connectionHosts[0].port || 9042;

  const clientOptions: ClientOptions = {
    host,
    port,
    contactPoints: [`${host}:${port}`],
  };

  if (database) {
    clientOptions.keyspace = database;
  }

  // client authentication
  if (connectionParameters?.username && connectionParameters?.password) {
    clientOptions.authProvider = {
      username: connectionParameters?.username,
      password: connectionParameters?.password,
    };
  }

  return clientOptions;
}