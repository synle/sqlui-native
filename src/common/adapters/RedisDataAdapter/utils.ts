import BaseDataAdapter from 'src/common/adapters/BaseDataAdapter/index';

export function getClientOptions(connectionOption: string) {
  const options = BaseDataAdapter.getConnectionParameters(connectionOption) as any;

  const { host, port } = options?.hosts[0];
  const { scheme, username, password } = options;

  const clientOptions: any = {
    url: `${scheme}://${host}:${port || 6379}`,
  };

  if (password) {
    clientOptions.password = password;
  }

  return clientOptions;
}