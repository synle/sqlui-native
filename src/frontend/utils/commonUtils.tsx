import { SqluiCore, SqluiFrontend } from 'typings';
// for exporting
export function getExportedConnection(connectionProps: SqluiCore.ConnectionProps) {
  const { id, connection, name } = connectionProps;
  return { _type: 'connection', ...{ id, connection, name } };
}

export function getExportedQuery(query: SqluiFrontend.ConnectionQuery) {
  const { id, name, sql, connectionId, databaseId, tableId } = query;
  return { _type: 'query', ...{ id, name, sql, connectionId, databaseId, tableId } };
}

// misc utils
const TO_BE_DELETED_LIST_ITEM = Symbol('to_be_deleted_list_item');

export function getUpdatedOrdersForList(items: any[], from: number, to: number) {
  if (from === to) {
    return items;
  }

  const targetItem = items[from];
  items[from] = TO_BE_DELETED_LIST_ITEM;

  // from > to : this is where we insert before `to`
  // from < to : this is where we insert after `to`
  items.splice(from > to ? to : to + 1, 0, targetItem);

  return items.filter((item) => item !== TO_BE_DELETED_LIST_ITEM);
}

export function getGeneratedRandomId(prefix: string) {
  return `${prefix}.${Date.now()}.${Math.floor(Math.random() * 10000000000000000)}`;
}

export async function createSystemNotification(message: string) {
  try {
    await Notification.requestPermission();
    new Notification(message);
  } catch (err) {}
}

export function sortColumnNamesForUnknownData(colNames: string[]) {
  return colNames.sort((a, b) => {
    // do sorting on columnname
    // attempt to show common column names (primary keys) first then sort other column names alphabetically
    const SPECIAL_COLUMN_NAMES = ['_id', 'id', 'rowKey', 'partitionKey', 'etag'];

    // here keep track of its position with respect to special column name
    let posa = SPECIAL_COLUMN_NAMES.indexOf(a);
    let posb = SPECIAL_COLUMN_NAMES.indexOf(b);
    posa = posa === -1 ? 100000 : posa;
    posb = posb === -1 ? 100000 : posb;

    // here keep track of its position if it has an id in name
    let ida = a.toLowerCase().endsWith('id') ? 0 : 1;
    let idb = b.toLowerCase().endsWith('id') ? 0 : 1;

    const sa = `${posa.toString().padStart(6, '0')}.${ida}.${a}`;
    const sb = `${posb.toString().padStart(6, '0')}.${idb}.${b}`;

    return sa.localeCompare(sb);
  });
}
