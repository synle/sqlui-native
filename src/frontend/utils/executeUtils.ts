// @ts-ignore
const { exec } =  window.requireElectron('child_process');

export function execute(shellToRun: string, delay = 25): Promise<string> {
  return new Promise((resolve, reject) => {
    setTimeout(() => {
      exec(shellToRun, (error, stdout, stderr) => {
        if (error) {
          return reject(stderr);
        }

        resolve(stdout);
      });
    }, delay);
  });
}
