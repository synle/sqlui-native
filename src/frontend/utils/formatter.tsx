import { format as _formatSQL } from 'sql-formatter';

const _formatJS = require('js-beautify').js;

export const formatSQL = (val: string) => {
  try {
    val = _formatSQL(val);
  } catch (err) {
    // if it's not working, let's remove all the leading space
    val = val
      .split('\n')
      .map((s) => s.trim())
      .join('\n');
  }
  return val;
};

export const formatJS = (val: string) => {
  try {
    val = _formatJS(val, {
      indent_size: 2,
      space_in_empty_paren: true,
      break_chained_methods: 2,
    });
  } catch (err) {}

  return val;
};

export const formatDuration = (durationMs: number) => {
  const durationS = Math.floor(durationMs / 1000);
  if (durationS > 1) return `${durationS} seconds`;

  return '<= 1 second';
};

export const escapeSQLValue = (value?: string) => {
  if (value === undefined) {
    value = '';
  }
  return value?.toString().replace(/'/g, `''`);
};

export const isValueNumber = (value: any) => {
  const parsed = parseFloat(value);
  return typeof parsed === 'number' && !isNaN(value);
};

export const isValueBoolean = (value: any) => {
  return (
    value === true ||
    value === false ||
    value.toString().toLowerCase() === 'true' ||
    value.toString().toLowerCase() === 'false'
  );
};