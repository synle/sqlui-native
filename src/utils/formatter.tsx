import { format as _formatSQL } from 'sql-formatter';
const _formatJS = require('js-beautify').js;

export const formatSQL = _formatSQL;
export const formatJS = (val: string) => _formatJS(val, {
          indent_size: 2,
          space_in_empty_paren: true,
          break_chained_methods: 2,
        });
