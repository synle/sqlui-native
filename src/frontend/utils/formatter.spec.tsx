import * as formatter from 'src/frontend/utils/formatter';

describe('formatter', () => {
  describe('formatJS', () => {
    test('should work', async () => {
      const actual = formatter.formatJS(
        `db.collection("sy-collection-1a").distinct("commute",{_id:"",commute:"","company.location":"","company.name":"",location:"",name:"",zip:""});`,
      );
      expect(actual).toMatchInlineSnapshot(
        `"db.collection(\\"sy-collection-1a\\").distinct(\\"commute\\",{_id:\\"\\",commute:\\"\\",\\"company.location\\":\\"\\",\\"company.name\\":\\"\\",location:\\"\\",name:\\"\\",zip:\\"\\"});"`,
      );
    });
  });

  describe('formatSQL', () => {
    test('should work', async () => {
      const actual = formatter.formatSQL(
        `SELECT AlbumId, ArtistId, Title FROM albums WHERE Title = 'abc' LIMIT 100`,
      );
      expect(actual).toMatchInlineSnapshot(`
"SELECT
  AlbumId,
  ArtistId,
  Title
FROM
  albums
WHERE
  Title = 'abc'
LIMIT
  100"
`);
    });
  });
});
