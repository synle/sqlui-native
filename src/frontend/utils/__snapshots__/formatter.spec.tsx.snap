// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatter formatJS should work 1`] = `
"db.collection(\\"sy-collection-1a\\")
  .distinct(\\"commute\\", {
    _id: \\"\\",
    commute: \\"\\",
    \\"company.location\\": \\"\\",
    \\"company.name\\": \\"\\",
    location: \\"\\",
    name: \\"\\",
    zip: \\"\\"
  });"
`;

exports[`formatter formatSQL should work 1`] = `
"SELECT
  AlbumId,
  ArtistId,
  Title
FROM
  albums
WHERE
  Title = 'abc'
LIMIT
  100"
`;
